NorwegianRefugeeCouncil/core
# Auth refactor

Jira ticket: JIRA-CORE24-#


## What have you implemented?
The Azure authorisation provider didn't allow the SCIM provisioning to authenticate. This PR removes the Azure auth provider and adds the auth handling to the express app.

Okta SCIM wants to use SAML or their own proprietary auth mechanism to authenticate. To use OIDC we have two apps in Okta. One is for SCIM that handles the user provisioning. The other is for OIDC that handles user auth. Access to both applications is handled through a single user group. Adding a user to this group will trigger the SCIM workflow and allow them to login.

I suspect there are issues here and it will need some future work (possibly to do with the frontend and redirect workflows).

This has caused the local oidc provider to not work. I am working on a fix for this but in the meantime use a dev Okta account.

* Setup passport.js
* Remove Azure auth
* Fix various things, eg. db connections, migrations, libs in docker image
* Move me controller into user controller
* Refactor auth middleware to use passport.js

## How to test it?
On core.app.nrc-dev.no, add/remove/change users and check that they are updated in core.app.nrc-dev.no/api/user

## Extra/next steps
* There is a postgres setting that I am unsure about. Needs DevOps input.
* Local database seeding isn't setup. Either we have it pull from Okta or wait until I have a working local oidc.
* I think we should move the user logic to a `user-engine` lib.
